### PR TITLE
server: double check host capacity when start/migrate a vm

### PR DESCRIPTION
**Pull request by ustcweizhou**
## Description

When start a vm or migrate a vm (away from a host in host maintenance), cloudstack will check capacity of all hosts and choose one. If there are hundreds of hosts on the platform, it will take some seconds. When cloudstack choose a host and start/migrate vm to it, the resource consumption of the host might have been changed. This normally happens when we start/migrate multiple vms.
It would be better to double check the host capacity when start vm on a host.

This PR includes the fix for cpucore capacity when start/migrate a vm.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

start multiple vms on a host
put a host to maintenance so multiple vms will be migrated at same time
check the resource count of cpucore, all looks good.

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
debian package failed:
@blueorangutan package

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java [hunk 1]
```diff
@@ -2349,6 +2349,17 @@ protected void migrate(final VMInstanceVO vm, final long srcHostId, final Deploy
...
+            _networkMgr.rollbackNicForMigration(vmSrc, profile);
+            s_logger.info("Migration cancelled because " + e2.getMessage());
+            work.setStep(Step.Done);
+            _workDao.update(work.getId(), work);
+            try {
+                stateTransitTo(vm, Event.OperationFailed, srcHostId);
+            } catch (final NoTransitionException e3) {
+                s_logger.warn(e3.getMessage());
+            }
+            throw new CloudRuntimeException("Migration cancelled because " + e2.getMessage());
```
Comment (DaanHoogland) [inline engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java]: can we add the cause or log its stack-trace before abandoning it here?

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by andrijapanicsb** [comment]
build failed, wiped the env.

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@andrijapanicsb this one looks good. Want to have a look? (cc @borisstoyanov )

**Comment by DaanHoogland** [comment]
@rhtyd @nvazquez @wido @GabrielBrascher is this alright?

**Pull request merged**